src/test/java/org/commcare/formplayer/junit/InitializeStaticsExtension.java

```diff
@@ -0,0 +1,29 @@
+package org.commcare.formplayer.junit;
+
+import org.commcare.formplayer.application.SQLiteProperties;
+import org.commcare.formplayer.engine.ClasspathFileRoot;
+import org.commcare.formplayer.util.PrototypeUtils;
+import org.javarosa.core.model.utils.DateUtils;
+import org.javarosa.core.reference.ReferenceManager;
+import org.javarosa.core.services.locale.Localization;
+import org.javarosa.core.services.locale.LocalizerManager;
+import org.junit.jupiter.api.extension.BeforeAllCallback;
+import org.junit.jupiter.api.extension.BeforeEachCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+public class InitializeStaticsExtension implements BeforeEachCallback {
+    @Override
+    public void beforeEach(ExtensionContext context) throws Exception {
+        LocalizerManager.setUseThreadLocalStrategy(true);
+        Localization.getGlobalLocalizerAdvanced().addAvailableLocale("default");
+        Localization.setLocale("default");
+        ReferenceManager.instance().addReferenceFactory(new ClasspathFileRoot());
+        Localization.registerLanguageReference("default",
+                "jr://springfile/formplayer_translatable_strings.txt");
+
+        PrototypeUtils.setupThreadLocalPrototypes();
+        new SQLiteProperties().setDataDir("testdbs/");
```

Comment:
I see that `SQLiteProperties` already uses the `sqlite.dataDir` property so we can just use that. I've pushed some more commits that updates the `GenerateSnapshotDatabases` test.

Incidentally, where does that test get executed. In normal test runs it's a noop but there is a gradle task to call it: https://github.com/dimagi/formplayer/blob/40110bf84bbf59af4816c03dbca5f4d82d2d77ca/build.gradle#L165-L172

But I don't know where that's called.

Reply:
I see it is referenced here: https://github.com/dimagi/formplayer/blob/73ef9df1261fb86502788b6cdb94f5bd7df2cf83/src/test/java/org/commcare/formplayer/tests/SnapshotTests.java#L93-L95

Reply:
Yeah it's supposed to run manually to update the db snapshots in case of app db changes. There is more info in Javadoc for the `GenerateSnapshotDatabases` class.